### PR TITLE
Flow var inputs fix

### PR DESCRIPTION
**Pull request by ilasalexandra**
Signed-off by: Alexandra Ilas alexandra.ilas@microfocus.com

### CODE REVIEW

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/ExecutableExecutionData.java [hunk 1]
```diff
@@ -138,7 +139,16 @@ public void startExecutable(@Param(ScoreLangConstants.EXECUTABLE_INPUTS_KEY) Lis
...
-
+            if (systemContext.containsKey(DEBUGGER_EXECUTABLE_INPUTS)) {
+                Map<String, Value> debuggerInputs =
+                        (Map<String, Value>) systemContext.remove(DEBUGGER_EXECUTABLE_INPUTS);
+                List<Input> newInputs = debuggerInputs.keySet().stream().map(key ->
+                        new Input.InputBuilder(key, debuggerInputs.get(key)).build()).collect(toList());
+                List<Input> updatedExecutableInputs = new ArrayList<>(newExecutableInputs.size() + newInputs.size());
+                updatedExecutableInputs.addAll(executableInputs);
+                updatedExecutableInputs.addAll(newInputs);
+                newExecutableInputs = updatedExecutableInputs;
```
Comment (alexbajzat) [inline cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/ExecutableExecutionData.java:150]: Why do we need the `updatedExecutableInputs` ? just `.addAll` in the `newExecutableInputs`

**Review by alexbajzat — CHANGES_REQUESTED** [review]

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/ExecutableExecutionData.java (hunk 1 above)
Comment (ilasalexandra) [reply cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/ExecutableExecutionData.java:150]: newExecutableInputs is immutable

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/env/ContextStack.java [hunk 2]
```diff
@@ -42,4 +44,18 @@ public Context popContext() {
     public Context peekContext() {
         return stack.peek();
     }
+
+    public void updateVariables(Map<String, Value> newVariables) {
+        if (this.stack.size() > 1) {
+            this.stack.removeFirst();
+        }
```
[the remark this answers is not in the document]
Comment (ilasalexandra) [reply cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/env/ContextStack.java]: removed this and added flag for pushing to context in ExecutableExecutionData

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/env/ContextStack.java [hunk 3]
```diff
@@ -42,4 +44,15 @@ public Context popContext() {
     public Context peekContext() {
         return stack.peek();
     }
+
+    public void updateVariables(Map<String, Value> newVariables) {
+        Context flowContext = this.stack.getLast();
+        newVariables.keySet().forEach(variable -> {
```
Comment (Razvan-Morar) [inline cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/env/ContextStack.java:50]: filter(variableKey ->flowContext.getVariable(variableKey)!=null).
forEach(variableKey->flowContext.putVariable(...))

Comment (Razvan-Morar) [reply cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/env/ContextStack.java:50]: getImmutableViewOfVariables() returns a new immutable map every time

**Review by Razvan-Morar — APPROVED** [review]

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/env/ContextStack.java (hunk 3 above)
Comment (ilasalexandra) [reply cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/env/ContextStack.java]: done

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/ExecutableExecutionData.java [hunk 4]
```diff
@@ -138,7 +139,16 @@ public void startExecutable(@Param(ScoreLangConstants.EXECUTABLE_INPUTS_KEY) Lis
 
             List<Input> newExecutableInputs =
                     addUserDefinedStepInputs(executableInputs, callArguments, promptArguments);
-
+            if (systemContext.containsKey(DEBUGGER_EXECUTABLE_INPUTS)) {
+                Map<String, Value> debuggerInputs =
+                        (Map<String, Value>) systemContext.remove(DEBUGGER_EXECUTABLE_INPUTS);
+                List<Input> newInputs = debuggerInputs.keySet().stream().map(key ->
+                        new Input.InputBuilder(key, debuggerInputs.get(key)).build()).collect(toList());
```
[the remark this answers is not in the document]
Comment (ilasalexandra) [reply cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/ExecutableExecutionData.java]: done

**Pull request merged**